pilot/pkg/networking/core/v1alpha3/envoyfilter/cluster_patch.go

```diff
@@ -90,11 +90,15 @@ func mergeTransportSocketCluster(c *cluster.Cluster, cp *model.EnvoyFilterConfig
 			// to merge it again
 			return true, nil
 		}
-	} else if cpValueCast.GetTransportSocket() != nil && c.GetTransportSocket() != nil {
-		if cpValueCast.GetTransportSocket().Name == c.GetTransportSocket().Name {
-			tsmPatch = c.GetTransportSocket()
-		} else {
-			// There is a name mismatch, so we cannot do a deep merge. Instead just replace the transport socket
+		// Now check if this matches if existing cluster's transport socket.
+		if c.GetTransportSocket() != nil {
+			if cpValueCast.GetTransportSocket().Name == c.GetTransportSocket().Name {
+				tsmPatch = c.GetTransportSocket()
+			}
```

Comment:
I would suggest, although in theory ` c.GetTransportSocket() != nil` and ·`len(c.GetTransportSocketMatches()) > 0` can not be true simultanously 
```
if cpValueCast.GetTransportSocket() != nil {
    if len(c.GetTransportSocketMatches()) > 0 {
    } else if c.GetTransportSocket() != nil{
    }
```

Reply:
Ok. Changed.